ARCHITECTURE.md

```diff
@@ -0,0 +1,134 @@
+
+# Other Bugsnag PHP Framework Libraries
+This section covers the other available Bugsnag notifiers for PHP frameworks and how they are implemented and connected to the main Bugsnag-PHP libary and each other.
+
+## [Bugsnag-PSR-Logger](https://github.com/bugsnag/bugsnag-psr-logger)
+This library implements the [PSR-3 Logger Interface](http://www.php-fig.org/psr/psr-3/) specification to enable users to attach Bugsnag into a standardised logging system. It consists of three classes:
+- `AbstractLogger` an abstract class which implements the PSR spec `LoggerInterface`, which requires a `log` method in its implementors
+- `BugsnagLogger` a logger class which extends the above class.  This logger will record `debug` or `info` logs as `breadcrumb` objects, and will notify Bugsnag of any other log type
+- `MultiLogger` again extends the `AbstractLogger`, but accepts and array of loggers in its construction, allowing other PSR compliant loggers to be used simultaneously with the `BugsnagLogger`
+
+## [Bugsnag-Wordpress](https://github.com/bugsnag/bugsnag-wordpress)
+This plugin for wordpress enables Bugsnag through the plugins menu of the wordpress site. It requires an older version of this library (~ 2.2) and so some of the methods and features will likely have been refactored for the newer versions.
+
+## [Bugsnag-Symfony](https://github.com/bugsnag/bugsnag-symfony)
+This library provides an extended version of the base Bugsnag-PHP library customized for the [Symfony PHP framework](symfony.com).
+
+### Dependency Management
+The libary is bundled for inclusion in the Symfony app framework through the `RegisterBundle` function used for adding Symfony extensions as laid out [here](https://symfony.com/doc/current/bundles.html).  It utilises the `DependencyInjection` folder to set up the extension through the `Configuration.php` file to define the necessary default configuration options, drawing the rest from the apps `config.yml`.  This configuration is then read in and processed by the `BugsnagExtension.php` file, and the arguments are set on the container.
+
+To build the client when requested, the `Configuration.php` defines a factory for the framework to use,  `ClientFactory.php`.  This factory wraps the creation methods for the `client` object and ensures that the client is configured correctly to get information from the framework and respond to events.  The `ClientFactory` configuration can be found in the Bugsnag service definition in `Resources/services.yml`.
+
+### Customizing the `Client` object
+The configuration passed through to the `ClientFactory` will modify several of the `client` objects properties.  In additional to the [configuration options](https://docs.bugsnag.com/platforms/php/symfony/configuration-options/) for the user to set-up their particular configuration, it also defines the `resolver` object to be used to gather information to populate the `report` objects used in the notify method.
```

Comment:
set-up -> setup
to be used to gather information to populate the -> which gathers information that populates